allure2-model-api/src/main/java/io/qameta/allure/FileSystemResultsWriter.java

```diff
@@ -37,7 +36,7 @@ public void write(TestResult testResult) {
                 : generateTestResultName(testResult.getUuid());
         createDirectories(outputDirectory);
         Path file = outputDirectory.resolve(testResultName);
-        try (OutputStream os = Files.newOutputStream(file, CREATE_NEW)) {
+        try (OutputStream os = new BufferedOutputStream(new FileOutputStream(file.toString()))) {
```

Comment:
why not using `new BufferedOutputStream(Files.newOutputStream(file, CREATE_NEW))` ?

Reply:
I debugged both operations, the former causes the allure results exeception. It's reproduced here: https://github.com/AutomatedOwl/allure-testng-result-write-bug

Reply:
I see, let me check it 

PS you need to add `aspectjweaver` dependency to maven surefire plugin in order to run it (I got no aspectjweaver:1.9.1 in my local maven repo, so build fails)

Reply:
Updated pom.xml at https://github.com/AutomatedOwl/allure-testng-result-write-bug

Reply:
Yeah, I guess we need to use blocking io, so your solution is fine. Please replace `file.toString()` with `file.toFile()` and update other methods `write` as well

Reply:
@baev changes done